src/Areas/Marketplace/Services/MarketplaceService.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            $"api-version={ApiVersion}"
+        };
+
+        if (includeStopSoldPlans.HasValue)
+            queryParams.Add($"includeStopSoldPlans={includeStopSoldPlans.Value.ToString().ToLower()}");
+
+        if (!string.IsNullOrEmpty(language))
+            queryParams.Add($"language={Uri.EscapeDataString(language)}");
+
+        if (!string.IsNullOrEmpty(market))
+            queryParams.Add($"market={Uri.EscapeDataString(market)}");
+
+        if (lookupOfferInTenantLevel.HasValue)
+            queryParams.Add($"lookupOfferInTenantLevel={lookupOfferInTenantLevel.Value.ToString().ToLower()}");
+
+        if (!string.IsNullOrEmpty(planId))
+            queryParams.Add($"planId={Uri.EscapeDataString(planId)}");
+
+        if (!string.IsNullOrEmpty(skuId))
+            queryParams.Add($"skuId={Uri.EscapeDataString(skuId)}");
+
+        if (includeServiceInstructionTemplates.HasValue)
+            queryParams.Add($"includeServiceInstructionTemplates={includeServiceInstructionTemplates.Value.ToString().ToLower()}");
+
+        string queryString = string.Join("&", queryParams);
+        return $"{ManagementApiBaseUrl}/subscriptions/{subscription}/providers/Microsoft.Marketplace/products/{productId}?{queryString}";
+    }
+
+    private async Task<string> GetMarketplaceResponseAsync(string url, string? partnerTenantId,
+        string? pricingAudience, string? tenant, RetryPolicyOptions? retryPolicy = null)
+    {
+        // Use Azure Core pipeline approach consistently
+        var clientOptions = AddDefaultPolicies(new MarketplaceClientOptions());
+
+        // Configure retry policy if provided
+        if (retryPolicy != null)
+        {
+            clientOptions.Retry.MaxRetries = retryPolicy.MaxRetries;
+            clientOptions.Retry.Mode = retryPolicy.Mode;
+            clientOptions.Retry.Delay = TimeSpan.FromSeconds(retryPolicy.DelaySeconds);
+            clientOptions.Retry.MaxDelay = TimeSpan.FromSeconds(retryPolicy.MaxDelaySeconds);
+            clientOptions.Retry.NetworkTimeout = TimeSpan.FromSeconds(retryPolicy.NetworkTimeoutSeconds);
+        }
+
+        // Create pipeline
+        var pipeline = HttpPipelineBuilder.Build(clientOptions);
+
+        string accessToken = await GetAccessTokenAsync(tenant);
+
+        var request = pipeline.CreateRequest();
+        request.Method = RequestMethod.Get;
+        request.Uri.Reset(new Uri(url));
+        request.Headers.Add("Authorization", $"Bearer {accessToken}");
+
+        // Add optional headers as specified in the API
+        if (!string.IsNullOrEmpty(partnerTenantId))
+            request.Headers.Add("PartnerTenantId", partnerTenantId);
+
+        if (!string.IsNullOrEmpty(pricingAudience))
+            request.Headers.Add("PricingAudience", pricingAudience);
+
+        var response = await pipeline.SendRequestAsync(request, CancellationToken.None);
+
+        if (!response.IsError)
+        {
+            return response.Content.ToString();
```

Comment:
I would expect a POJO like ProductSummary[] to be output from this rather than a string.